table/iterator.go

```diff
@@ -204,10 +204,30 @@ func (itr *Iterator) seekHelper(blockIdx int, key []byte) {
 		return
 	}
 	itr.bi.setBlock(block)
-	itr.bi.seek(key)
+	itr.bi.seek(key, 0)
 	itr.err = itr.bi.Error()
 }
 
+const seekBound = 3
+
+func (itr *Iterator) seekFromOffset(blockIdx int, offset int, key []byte) {
+	itr.bpos = blockIdx
+	block, err := itr.t.block(blockIdx)
+	if err != nil {
+		itr.err = err
+		return
+	}
+	itr.bi.setBlock(block)
+	for i := 0; i < seekBound; i++ {
```

Comment:
Why do we need the `seekBound`?

Reply:
Badger encoding a timestamp in the end of key. It is possible there are multi version of a key stored in this block. 

In this case the position in hash index is point to the latest version, so we need to find the appropriate version of the request. 

Because the versions of the key may not be much, we can do a few next to check them, instead of do a binary search.

Reply:
Can we just fallback if we got the newer version key?
This is a very rare case, maybe less than 1%?
Remove this logic can simplify the code a lot.

Reply:
Got it.